optaplanner-core/src/main/java/org/optaplanner/core/api/score/constraint/ConstraintJustificationComparator.java

```diff
@@ -0,0 +1,37 @@
+package org.optaplanner.core.api.score.constraint;
+
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.optaplanner.core.config.util.ConfigUtils;
+import org.optaplanner.core.impl.domain.common.accessor.MemberAccessor;
+
+final class ConstraintJustificationComparator implements Comparator<Object> {
+
+    private static Comparator<Object> CLASS_NAME_COMPARATOR = Comparator.comparing(o -> o.getClass().getCanonicalName());
+    private final Map<Class<?>, MemberAccessor> accessorCache = new HashMap<>(0);
+
+    @Override
+    public int compare(Object o1, Object o2) {
```

Comment:
This is the meat of the PR. This is how all constraint justification objects would be compared.

Reply:
This also probably deserves some tests. Will add.